scripts/core/EXT_Exp_GroupAlgorithmMemory.rst

```diff
@@ -0,0 +1,34 @@
+<%
+import re
+from templates import helper as th
+%><%
+    OneApi=tags['$OneApi']
+    x=tags['$x']
+    X=x.upper()
+%>
+:orphan:
+
+.. _ZE_experimental_device_group_algorithm_memory_properties:
+
+========================================================
+Group Algorithm Memory Properties Experimental Extension
+========================================================
+
+API
+----
+
+* Enumerations
+
+    * ${x}_device_group_algorithm_memory_properties_exp_version_t
+    * ${x}_group_algorithm_type_exp_t
+    * ${x}_group_algorithm_memory_scope_exp_t
+
+* Structures
+
+
+    * ${x}_device_group_algorithm_memory_exp_properties_t
+
+* Functions
+
+
+    * ${x}DeviceGetGroupAlgorithmMemoryPropertiesExp
```

Comment:
Would it beneficial to have an explanation of this API with an example usage after the API overview section?

Reply:
@andreyfe1 : Please let me know if the following text is correct:

```cpp
This extension allows applications to query for memory requirements of
the underlying compiler when selecting a particular group algorithm in
their kernels. Three types of algorithms are available:

$X_GROUP_ALGORITHM_TYPE_EXP_SORT: sorting algorithm

$X__GROUP_ALGORITHM_TYPE_EXP_SCAN: scan algorithm

$X__GROUP_ALGORITHM_TYPE_EXP_REDUCE: reduce algorithm

To know what amount of memory is needed to run use algorithms,
application would call this extension as follows:


$x_device_group_algorithm_memory_exp_properties_t groupAlgorithmMemoryProperties {];
groupAlgorithmMemoryProperties.stype = $X_STRUCTURE_TYPE_DEVICE_GROUP_ALGORITHM_MEMORY_EXP_PROPERTIES;
groupAlgorithmMemoryProperties.algorithm = $X_GROUP_ALGORITHM_TYPE_EXP_SORT;
groupAlgorithmMemoryProperties.memoryScope = $X_GROUP_ALGORITHM_MEMORY_SCOPE_EXP_SUBGROUP;

$x_result_t res = $xDeviceGetGroupAlgorithmMemoryPropertiesExp(hDevice, numOfElements, &groupAlgorithmMemoryProperties);

Upon return, groupAlgorithmMemoryProperties will contain in
globalMemorySize and sharedLocalMemorySize the global and
shared local memory needed, respectively, for executing the selected
algorithm in the target device.

```

Reply:
@jandres742 Is the double underscore intentional? i.e. should `$X__` be `$X_` ?

Reply:
@wdamon-intel thanks. that's just an error. Once I write it in the scripts, I will correct it. Text in comment above is just a draft to confirm it is good.

Reply:
> This extension allows applications to query for memory requirements of the underlying compiler when selecting a particular group algorithm in their kernels

@jandres742, the goal for this extension is to let users know how much temporary memory they need to allocate. When they know the value, they can allocate a memory and put it as a parameter to a group algorithm that requires additional memory. So, regarding wording. Maybe we can say something like this
```
This extension allows applications to query for memory requirements of
the underlying compiler when selecting a particular group algorithm in
their kernels. It gives applications the memory size that is required for
memory allocations needed for group algorithms.
```